### PR TITLE
intial test setup

### PR DESCRIPTION
**Pull request by vinayak-trivedi**
### What is the change?

Added the initial test setup. solves #34

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -57,6 +62,16 @@
     "tslint-config-prettier": "^1.18.0",
     "typescript": "^4.1.2"
   },
+  "jest": {
+    "coverageThreshold": {
+      "global": {
+        "branches": 85,
+        "functions": 85,
+        "lines": 85,
+        "statements": -10
```
Comment (harshith-venkatesh) [inline package.json:71]: Why is this -10?

File: package.json [hunk 2]
```diff
@@ -57,6 +62,16 @@
     "tslint-config-prettier": "^1.18.0",
     "typescript": "^4.1.2"
   },
+  "jest": {
+    "coverageThreshold": {
+      "global": {
+        "branches": 85,
+        "functions": 85,
+        "lines": 85,
```
Comment (harshith-venkatesh) [inline package.json:70]: Could we focus on 95+ from the beginning itself?

**Review by harshith-venkatesh — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (vinayak-trivedi) [reply package.json:71]: Statement -10 ensures that the test fails  if there are more than 10 uncovered statements

File: package.json (hunk 2 above)
Comment (vinayak-trivedi) [reply package.json:70]: yes sure I will do that

File: package.json (hunk 1 above)
Comment (harshith-venkatesh) [reply package.json:71]: Could you share the article for the same

Comment (vinayak-trivedi) [reply package.json:71]: https://jestjs.io/docs/configuration#coveragethreshold-object

Comment (harshith-venkatesh) [reply package.json:71]: It is bit confusing I feel, Could we go ahead with 95% here as well?

Comment (vinayak-trivedi) [reply package.json:71]: This line can be removed then

Comment (harshith-venkatesh) [reply package.json:71]: Do you feel removing the statements coverage altogether a good option or could we go ahead with 95% here as well

**Comment by harshith-venkatesh** [comment]
Rest all looks good, I will merge the PR once addressed.
Good work @vinayak-trivedi

File: package.json [hunk 3]
```diff
@@ -57,6 +62,15 @@
     "tslint-config-prettier": "^1.18.0",
     "typescript": "^4.1.2"
   },
+  "jest": {
+    "coverageThreshold": {
+      "global": {
+        "branches": 95,
+        "functions": 95,
+        "lines": 95
```
Comment (harshith-venkatesh) [inline package.json:70]: We do need statements coverage

File: package.json (hunk 1 above)
Comment (vinayak-trivedi) [reply package.json]: Basically it is specified as a negative number to represent the maximum number of uncovered entities allowed.

File: package.json (hunk 3 above)
Comment (vinayak-trivedi) [reply package.json]: added

**Review by harshith-venkatesh — APPROVED** [review]

**Pull request merged**